src/mame/audio/supracan_um6619_audiosoc.h

```diff
@@ -2,26 +2,24 @@
 // copyright-holders:Ryan Holtz, superctr
 /**********************************************************************
 
-    Super A'Can sound driver
+    Super A'Can sound driver (UM6619)
 
 **********************************************************************/
 
-#ifndef MAME_AUDIO_ACAN_H
-#define MAME_AUDIO_ACAN_H
+#ifndef MAME_AUDIO_SUPRACAN_UM6619_AUDIOSOC_H
+#define MAME_AUDIO_SUPRACAN_UM6619_AUDIOSOC_H
 
 #pragma once
 
-class acan_sound_device : public device_t, public device_sound_interface
+#include "supracan_um6619_cpu.h"
+
+class supracan_um6619_audiosoc : public supracan_um6619_cpu_device, public device_sound_interface
```

Comment:
Deriving this from the CPU device doesn’t seem like a great design choice – it doesn’t really do much for encapsulation.  It would be better if it were a device that the CPU could instantiate.

Reply:
other SoCs such as the sunplus ones derive from the CPU class, I was copying that.